src/lib/shell/commands/Ota.cpp

```diff
@@ -73,6 +73,74 @@ CHIP_ERROR NotifyImageHandler(int argc, char ** argv)
     return CHIP_NO_ERROR;
 }
 
+CHIP_ERROR StateHandler(int argc, char ** argv)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    VerifyOrReturnError(GetRequestorInstance() != nullptr, CHIP_ERROR_INCORRECT_STATE);
+    VerifyOrReturnError(argc == 0, CHIP_ERROR_INVALID_ARGUMENT);
+
+    app::Clusters::OtaSoftwareUpdateRequestor::OTAUpdateStateEnum state;
+    err = GetRequestorInstance()->GetState(0, state);
```

Comment:
Overall, is the shell running within the Matter event loop? If not, there is potentially cross-thread access here that may yield corrupt data if called during the download.

Reply:
I added scheduling tasks for the PlatformMgr() to not handle it outside of the Matter event loop.